packages/react-dom/src/events/ReactDOMEventListener.js

```diff
@@ -125,70 +155,89 @@ export function isEnabled() {
   return _enabled;
 }
 
-/**
- * Traps top-level events by using event bubbling.
- *
- * @param {number} topLevelType Number from `TopLevelEventTypes`.
- * @param {object} element Element on which to attach listener.
- * @return {?object} An object with a remove function which will forcefully
- *                  remove the listener.
- * @internal
- */
 export function trapBubbledEvent(
   topLevelType: DOMTopLevelEventType,
-  element: Document | Element,
-) {
-  if (!element) {
-    return null;
-  }
-  const dispatch = isInteractiveTopLevelEventType(topLevelType)
-    ? dispatchInteractiveEvent
-    : dispatchEvent;
-
-  addEventBubbleListener(
-    element,
-    getRawEventName(topLevelType),
-    // Check if interactive and wrap in interactiveUpdates
-    dispatch.bind(null, topLevelType),
-  );
+  element: Document | Element | Node,
+): void {
+  trapEventForPluginEventSystem(element, topLevelType, false);
 }
 
-/**
- * Traps a top-level event by using event capturing.
- *
- * @param {number} topLevelType Number from `TopLevelEventTypes`.
- * @param {object} element Element on which to attach listener.
- * @return {?object} An object with a remove function which will forcefully
- *                  remove the listener.
- * @internal
- */
 export function trapCapturedEvent(
   topLevelType: DOMTopLevelEventType,
-  element: Document | Element,
-) {
-  if (!element) {
-    return null;
+  element: Document | Element | Node,
+): void {
+  trapEventForPluginEventSystem(element, topLevelType, true);
+}
+
+export function trapEventForResponderEventSystem(
+  element: Document | Element | Node,
+  topLevelType: DOMTopLevelEventType,
+  capture: boolean,
+  passive: boolean,
+): void {
+  if (enableEventAPI) {
+    const dispatch = isInteractiveTopLevelEventType(topLevelType)
+      ? dispatchInteractiveEvent
+      : dispatchEvent;
+    const rawEventName = getRawEventName(topLevelType);
+    let eventFlags = RESPONDER_EVENT_SYSTEM;
+
+    // If passive option is not supported, then the event will be
+    // active and not passive, but we flag it as using not being
+    // supported too. This way the responder event plugins know,
+    // and can provide polyfills if needed.
+    if (passive) {
+      if (passiveBrowserEventsSupported) {
```

Comment:
This seems like the wrong check to me. Shouldn't this case be inverted? This will mark it as active and PASSIVE_NOT_SUPPORTED only if passive events *are* supported.

Reply:
Well spotted!